gtsam/hybrid/HybridBayesNet.h

```diff
@@ -101,6 +87,35 @@ class GTSAM_EXPORT HybridBayesNet : public BayesNet<HybridConditional> {
         std::make_shared<HybridConditional>(std::move(conditional)));
   }
 
+  /**
+   * @brief Add a conditional to the Bayes net.
+   * Implicitly convert to a HybridConditional.
+   *
+   * E.g.
+   * hbn.push_back(std::make_shared<DiscreteConditional>(m, "1/1"));
+   *
+   * @tparam CONDITIONAL Type of conditional. This is shared_ptr version.
+   * @param conditional The conditional as a shared pointer.
+   */
+  template <class CONDITIONAL>
+  void push_back(std::shared_ptr<CONDITIONAL> conditional) {
+    factors_.push_back(std::make_shared<HybridConditional>(conditional));
+  }
+
+  /**
+   * Preferred: add a conditional directly using a pointer.
```

Comment:
Hmmm, no, emplace_shared<T> should be preferred. I don’t think this raw pointer call fits anything we have ever done in GTSAM.

Reply:
This was pre-existing and confusing to me as well. I'm going to remove it.